src/compute-client/src/controller/replicated.rs

```diff
@@ -389,6 +391,39 @@ struct ReplicaState<T> {
     response_rx: UnboundedReceiver<ComputeResponse<T>>,
     /// The network addresses of the processes that make up the replica.
     addrs: Vec<String>,
+    /// Where to persist introspection sources
+    persisted_logs: BTreeMap<LogVariant, (GlobalId, CollectionMetadata)>,
+}
+
+impl<T> ReplicaState<T> {
+    /// Specialize a command for the given `Replica` and `ReplicaId`.
+    ///
+    /// Most `ComputeCommand`s are independent of the target replica, but some
+    /// contain replica-specific fields that must be adjusted before sending.
+    fn specialize_command(&self, command: &mut ComputeCommand<T>, replica_id: ReplicaId) {
+        // Tell new instances their replica ID.
```

Comment:
This comment is a bit misplaced now, since the block below does more than setting the replica ID.